client/src/lib/utils/ui.ts

```diff
@@ -0,0 +1,12 @@
+export const handleSmoothScroll = (e: React.MouseEvent<HTMLAnchorElement>) => {
```

Comment:
You don't need JS to do this 😉: https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior

Reply:
Sadly I couldn't make this work with Next 13